x-pack/plugins/advanced_ui_actions/public/drilldowns/drilldown_definition.ts

```diff
@@ -0,0 +1,114 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ActionFactoryDefinition } from '../dynamic_actions';
+import { DrilldownActionFactoryContext } from './drilldown_action_factory_context';
+
+/**
+ * This is a convenience interface to register a drilldown. Drilldown has
+ * ability to collect configuration from user. Once drilldown is executed it
+ * receives the collected information together with the context of the
+ * user's interaction.
+ *
+ * `Config` is a serializable object containing the configuration that the
+ * drilldown is able to collect using UI.
+ *
+ * `PlaceContext` is an object that the app that opens drilldown management
+ * flyout provides to the React component, specifying the contextual information
+ * about that app. For example, on Dashboard app this context contains
+ * information about the current embeddable and dashboard.
+ *
+ * `ExecutionContext` is an object created in response to user's interaction
+ * and provided to the `execute` function of the drilldown. This object contains
+ * information about the action user performed.
+ */
+export interface DrilldownDefinition<
+  Config extends object = object,
+  PlaceContext extends object = object,
+  ExecutionContext extends object = object
+> {
+  /**
+   * Globally unique identifier for this drilldown.
+   */
+  id: string;
+
+  /**
+   * Determines the display order of the drilldowns in the flyout picker.
+   * Higher numbers are displayed first.
+   */
+  order?: number;
+
+  /**
+   * Function that returns default config for this drilldown.
+   */
+  createConfig: ActionFactoryDefinition<
+    Config,
+    DrilldownActionFactoryContext<PlaceContext>,
+    ExecutionContext
+  >['createConfig'];
+
+  /**
+   * `UiComponent` that collections config for this drilldown. You can create
+   * a React component and transform it `UiComponent` using `uiToReactComponent`
+   * helper from `kibana_utils` plugin.
+   *
+   * ```tsx
+   * import React from 'react';
+   * import { uiToReactComponent } from 'src/plugins/kibana_utils';
+   * import { CollectConfigProps } from 'src/plugins/kibana_utils/public';
+   *
+   * type Props = CollectConfigProps<Config>;
+   *
+   * const ReactCollectConfig: React.FC<Props> = () => {
+   *   return <div>Collecting config...'</div>;
+   * };
+   *
+   * export const CollectConfig = uiToReactComponent(ReactCollectConfig);
+   * ```
+   */
+  CollectConfig: ActionFactoryDefinition<
+    Config,
+    DrilldownActionFactoryContext<PlaceContext>,
+    ExecutionContext
+  >['CollectConfig'];
+
+  /**
+   * A validator function for the config object. Should always return a boolean
+   * given any input.
+   */
+  isConfigValid: ActionFactoryDefinition<
+    Config,
+    DrilldownActionFactoryContext<PlaceContext>,
+    ExecutionContext
+  >['isConfigValid'];
```

Comment:
I would rather keep it as it is, because drilldown is built on top of action factories, not configurable; and any type changes in action factory would automatically reflect here.